### PR TITLE
init idea for clean unneeded files

### PR DESCRIPTION
**Pull request by YifeiZuo01**
### Status:
*READY*

### Description

This is the proposal for remove files other than DEATIL.TXT and MBED.html for K64F currently.

Updates:
Used into all tests.

### CODE REVIEW

**Comment by YifeiZuo01** [comment]
move two self.assert after files get removed, otherwise, assert failure would stop file get cleaned.

**Comment by YifeiZuo01** [comment]
any comment?

File: test/test_helper.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    @staticmethod
+    def del_file(file_path, file_name):
+        """
+        :param file_path: file path
+        :param file_name: file name of file to be deleted
+        :return:
+        """
+        if os.path.isfile(os.path.join(file_path, file_name)):
+            if platform.system() == 'Windows':
+                os.system('del /F %s' % os.path.join(file_path, file_name))
```
Comment (juhhov) [inline test/test_helper.py]: Why not use os.remove?

**Comment by juhhov** [comment]
How much longer does it take to run the tests with this?

**Comment by YifeiZuo01** [comment]
@juhhov  1 min in total

data got from CI console log, without checking file and delete, it's 6 mins, after this changes it's 7.

File: test/test_helper.py (hunk 1 above)
Comment (YifeiZuo01) [reply test/test_helper.py]: now it's 6 mins, no big difference with os.remove

**Comment by YifeiZuo01** [comment]
feels like  (linux )tests are more stable now,  @VeliMattiLahtela  can we bring devices back to windows?

File: test/test_helper.py (hunk 1 above)
Comment (YifeiZuo01) [reply test/test_helper.py]: now we're facing issue of deleting windows "System Volume Information"
@juhhov  suggested to just delete FAIL.txt and ASSERT.TXT
@jupe  how do think?
````
10:10:13 [windows]     Helper(platform_name='K64F', allowed_files=['DETAILS.TXT', 'MBED.HTM']).clear()
10:10:13 [windows]   File "C:\Jenkins\ws\mbed-flasher_PR-70-QZD4OYH22FQ2ZZO2T2SBCPXY6APY2\mbed-flasher\test\test_helper.py", line 44, in clear
10:10:13 [windows]     os.remove(os.path.join(mount_point, bad_file))
10:10:13 [windows] PermissionError: [WinError 5] Access is denied: 'D:System Volume Information'
```

Comment (jupe) [reply test/test_helper.py]: seems that bad_file points to invalid place... something wrong in code side I think

Comment (YifeiZuo01) [reply test/test_helper.py]:
```C:\>dir /a:hd D:
 Volume in drive D is DAPLINK
 Volume Serial Number is 2702-1974
 Directory of D:\
13.10.2017  12.08    <DIR>          System Volume Information
               0 File(s)              0 bytes
               1 Dir(s)      67 088 384 bytes free
```

Comment (YifeiZuo01) [reply test/test_helper.py]: This should be the correct place. windows seems create this file for each USB connected device

Comment (YifeiZuo01) [reply test/test_helper.py]: yes, you're right, this is a bug, fixed now. :)

**Comment by YifeiZuo01** [comment]
everything is ready, please review :)

**Comment by jonikula** [comment]
Should update to newest master before merging.

File: test/test_helper.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        if self.platform_name:
+            targets = self._get_targets()
+
+            # Clear files other than allowed
+            for target in targets:
+                mount_point = target.get('mount_point')
+                if mount_point:
+                    for bad_file in os.listdir(mount_point):
+                        if os.path.isfile(bad_file) and bad_file not in self.allowed_files:
+                            os.remove(os.path.join(mount_point, bad_file))
```
Comment (jonikula) [inline test/test_helper.py]: Should there be some handling for possible errors, like missing permissions to handle the files?

Comment (juhhov) [reply test/test_helper.py]: setup/teardown will fail if no permissions, i think that is ok

**Pull request merged**